spring-cloud-generator/scripts/generate-showcase.sh

```diff
@@ -1,111 +1,175 @@
+spring_root_dir=${SPRING_GENERATOR_DIR}/..
+showcase_starter_old_dir=${SPRING_GENERATOR_DIR}/showcase/showcase-spring-starter
+showcase_starter_new_dir=${SPRING_GENERATOR_DIR}/showcase/showcase-spring-starter-generated
 
 # Verifies newly generated showcase-spring-starter against goldens
 #
 # $1 - directory containing existing showcase-spring-starter (golden)
 # $2 - directory containing newly generated showcase-spring-starter
 function verify(){
-  OLD_DIR=$1
-  NEW_DIR=$2
-  SHOWCASE_STARTER_DIFF=$(diff -r ${NEW_DIR}/src/main ${OLD_DIR}/src/main)
-  SHOWCASE_STARTER_POM_DIFF=$(diff -r ${NEW_DIR}/pom.xml ${OLD_DIR}/pom.xml)
-  if [ "$SHOWCASE_STARTER_DIFF" != "" ] || [ "$SHOWCASE_STARTER_POM_DIFF" != "" ]
+  old_dir=$1
+  new_dir=$2
+  showcase_starter_diff=$(diff -r ${new_dir}/src/main ${old_dir}/src/main)
+  showcase_starter_pom_diff=$(diff -r ${new_dir}/pom.xml ${old_dir}/pom.xml)
+  if [ "${showcase_starter_diff}" != "" ] || [ "${showcase_starter_pom_diff}" != "" ]
   then
       echo "Differences detected in generated showcase starter module: "
       echo "Diff from src/main: "
-      echo $SHOWCASE_STARTER_DIFF
+      echo "${showcase_starter_diff}"
       echo "Diff from pom.xml: "
-      echo $SHOWCASE_STARTER_POM_DIFF
+      echo "${showcase_starter_pom_diff}"
       exit 1;
   else
       echo "No differences found in showcase-spring-starter"
-      rm -r ${NEW_DIR}
+      rm -r "${new_dir}"
   fi
 }
 
 # Setup, generation, and post-processing steps for showcase-spring-starter
 #
 # $1 - target directory for generated starter
 function generate_showcase_spring_starter(){
-  SHOWCASE_STARTER_DIR=$1
+  showcase_starter_dir=$1
 
   # Compute the parent project version.
-  cd ${SPRING_ROOT_DIR}
-  PROJECT_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
-  cd ${SPRING_GENERATOR_DIR}
-  GAPIC_GENERATOR_JAVA_VERSION=$(mvn help:evaluate -Dexpression=gapic-generator-java-bom.version -q -DforceStdout)
+  pushd "${spring_root_dir}"
+  export project_version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+  cd "${SPRING_GENERATOR_DIR}"
+  gapic_generator_java_version=$(mvn help:evaluate -Dexpression=gapic-generator-java-bom.version -q -DforceStdout)
 
-  if [[ -z "$GAPIC_GENERATOR_JAVA_VERSION" ]]; then
+  if [[ -z "${gapic_generator_java_version}" ]]; then
     echo "Missing sdk-platform-java commitish to checkout"
     exit 1
   fi
 
   # Clone sdk-platform-java (with showcase library)
-  git clone https://github.com/googleapis/sdk-platform-java.git
-  cd sdk-platform-java && git checkout "v${GAPIC_GENERATOR_JAVA_VERSION}"
+  if [[ ! -d "./sdk-platform-java" ]]; then
+    git clone https://github.com/googleapis/sdk-platform-java.git
+  fi
+  pushd sdk-platform-java
+  git checkout "v${gapic_generator_java_version}"
+
+  # We will use the generation tools from library_generation
+  pushd library_generation/utils
+  source utilities.sh
```

Comment:
How many utility methods in `utilities.sh` are we relying on? If it's not too much, I would prefer just duplicate the logic, because this creates dependencies of our hermetic build scripts, and we may accidentally break SpringCodeGen scripts.

Reply:
I'm moving them to `generate-showcase-utilities`